src/mscorlib/src/System/Span.cs

```diff
@@ -420,5 +439,49 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void Fill<T>(ref T destination, T value, int elementsCount)
+        {
+            if (!JitHelpers.ContainsReferences<T>())
+            {
+                int size = Unsafe.SizeOf<T>();
+
+                int copiedCount; // Fill up to copiedCount in first loop
+                if (size <= sizeof(int))
+                    copiedCount = elementsCount <= 512 / 4 * 3 ? elementsCount : 512 / 4 * 2;
```

Comment:
I see that Memmove use 512 as magic const when it make copy in loop or call system function.